apps/dtlaunch/app-b2.js

```diff
@@ -99,13 +99,31 @@
 
   Bangle.drawWidgets(); // To immediately update widget field to follow current theme - remove leftovers if previous app set custom theme.
   Bangle.loadWidgets();
-  drawPage(0);
+  drawPage(page);
 
-  for (let i = 4; i < apps.length; i++) { // Load the rest of the app icons that were not initially.
+  for (let i = 0; i < apps.length; i++) { // Load the rest of the app icons that were not initially.
+    if (i >= page*4 && i < Math.min(page*4+4, apps.length)) continue;
     if (apps[i].icon)
       apps[i].icon = s.read(apps[i].icon); // should just be a link to a memory area
   }
 
+  if (!Bangle.dtHandlePagePersist) {
+    Bangle.dtHandlePagePersist = (page) => {
+      // Function for persisting the active page when leaving dtlaunch.
+      if (page===undefined) {return this.page||0;}
+
+      if (!this.killHandler) { // Only register kill listener once.
+        this.killHandler = () => {
+          s.write("dtlaunch.page", this.page.toString());
+        };
+        E.on("kill", this.killHandler); // This is intentionally left around after fastloading into other apps. I.e. not removed in uiRemove.
+      }
+
+      this.page = page;
+    };
+    Bangle.dtHandlePagePersist(page);
```

Comment:
Similarly to the `this` thing above, do we want to chuck this onto `Bangle`, or perhaps we just have an unrelated global object we use instead?

Reply:
Maybe that's better!

Should I do it by e.g. `global.dtlaunch = {}; global.dtlaunch.handlePagePersist = ()=>{...}`?

Reply:
Nice, yeah I think that'll work great

Reply:
Nice - perhaps we want to camelCase the function to keep with the existing pattern? (`handlePagePersist` instead of `HandlePagePersist`)

Reply:
Yes - just noticed that as well 👍

Reply:
> Nice, yeah I think that'll work great - and if you make it
> ```javascript
> global.dtlaunch.handlePagePersist = function() {...}
> //                                  ^~~~~~~~
> ```
> 
> then when you call it as `dtlaunch.handlePagePersist()`, then you can use `this` exactly as you intended (as `this == dtlaunch`)

Interesting! So that's a difference between `function() {...} ` and `() => {...} `?

Reply:
Yes exactly - `() => ...` won't have its own `this`, it just uses one from up a level, whereas `function`s always have their own `this`. Also no, the `global` at the start of my suggestion isn't needed, nice spot

What you have now looks great btw!